app/models/extracts_for_classification.rb

```diff
@@ -11,10 +11,18 @@ def initialize(extracts)
     @extracts = extracts
   end
 
+  def empty?
+    extracts.empty?
+  end
+
   def select(&block)
     self.class.new(extracts.select(&block))
   end
 
+  def reject(&block)
+    self.class.new(extracts.reject(&block))
```

Comment:
I wanted to follow the same structure as the other methods on the class, and I do think that it's good to stick with non-destructive edits where possible.

Reply:
Was talking about them too. It doesn't seem wantonly destructive to remove extracts from a list you're trying to filter and are not going to use again. I'm 2/10 on it though.

Reply:
The same collection is, I think, being reused to provide subsets of the data to different reducers, so I don't want any chance of modifying the base collection, as it will introduce difficult-to-trace bugs. I feel pretty strongly about this one.

Reply:
Every time `#process` is called on a reducer, it calls `filter_extracts`, then the Reducer `extract_filter` model method, which creates a new instance of ExtractFilter. The grouped extracts passed to the filter isn't altered. Don't know what base collection you mean. 🤷‍♀️

Reply:
I don't intend to make this change.

Reply:
I don't feel strongly about this potential little refactor, but I am curious about your reasoning on this because, as I said, I don't understand which base collection you believe is at risk here. If you think I'm wrong about how this pathway works, fine, but suggestions and questions are what a PR review is for.